paddle/phi/kernels/onednn/softmax_kernel.cc

```diff
@@ -51,11 +50,38 @@ void SoftmaxKernel(const Context& dev_ctx,
       val = std::max(val, static_cast<T>(exp(-64)));
     });
   }
-
   out->set_mem_desc(dst_memory_p->get_desc());
 }
 
+template <typename T, typename Context>
+void SoftmaxKernel(const Context& dev_ctx,
+                   const DenseTensor& x,
+                   int axis,
+                   DenseTensor* out) {
+  bool is_inplace = x.IsSharedBufferWith(*out);
+
+  if (phi::funcs::is_int8<T>()) {
+    PADDLE_ENFORCE(!is_inplace,
+                   phi::errors::Unimplemented(
+                       "Inplace not implemeneted for int8 onednn softmax."));
+    const float output_scale = x.dtype() == DataType::INT8 ? 126.99f : 255.0f;
```

Comment:
Since this line only depends on x, what do you think about moving the scale calculation into the handler instead of passing it as argument?